### PR TITLE
COMMERCE-11440 Store Front handling of Products Priced as Price on Application

### PR DESCRIPTION
**Pull request by smottal**
https://issues.liferay.com/browse/COMMERCE-11440

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/Wrapper.js [hunk 1]
```diff
@@ -39,24 +40,25 @@ function Wrapper() {
...
+			{(requestQuoteEnabled || hasPriceOnApplication(cartItems)) &&
+				!!cartItems.length && (
+					<div className="request-quote-wrapper">
+						<ClayButton
+							block={true}
+							className="btn-md request-quote"
+							displayType="secondary"
+							onClick={() => {
+								return liferayNavigate(orderDetailURL);
+							}}
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/Wrapper.js:52]: Move this to a new request quote button component

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/OrderButton.js [hunk 2]
```diff
@@ -41,7 +41,7 @@ function OrderButton() {
 		<div className="mini-cart-submit">
 			<ClayButton
 				block
-				disabled={!cartItems.length}
+				disabled={!cartItems.length || hasPriceOnApplication(cartItems)}
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/OrderButton.js:44]: move disable logic to a new parameter of this component

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemsList.js [hunk 3]
```diff
@@ -37,6 +39,19 @@ export default function CartItemsList() {
 
 			{cartState.accountId && !!id && <CartQuickAdd />}
 
+			{hasPriceOnApplication(cartItems) && (
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemsList.js:42]: move this to  parameter of this component

File: modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/frontend/data/set/provider/PreviewCommerceOrderItemFDSDataProvider.java [hunk 4]
```diff
@@ -134,6 +134,28 @@ public List<PreviewOrderItem> getItems(
 				CommerceOrderItemPrice commerceOrderItemPrice =
 					commerceOrderImporterItem.getCommerceOrderItemPrice();
 
+				if ((commerceOrderItemPrice != null) &&
+					commerceOrderItemPrice.isPriceOnApplication()) {
+
+					return new PreviewOrderItem(
+						externalReferenceCode,
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/frontend/data/set/provider/PreviewCommerceOrderItemFDSDataProvider.java:141]: reviewing the wireframe we found out that the columns promoprice, discounts and total should render the dash '-' character

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java [hunk 5]
```diff
@@ -319,8 +319,18 @@ private Price _getPrice(
...
 			{
 				currency = commerceCurrency.getName(locale);
-				price = unitPrice.doubleValue();
 				priceFormatted = unitPriceCommerceMoney.format(locale);
+				priceOnApplication =
+					commerceProductPrice.isPriceOnApplication();
+
+				setPrice(
+					() -> {
+						if (unitPrice != null) {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java:326]: Please check if this is still needed unitPrice should be 0 not null

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
please review my comments

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/Wrapper.js (hunk 1 above)
Comment (smottal) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/Wrapper.js]: Done

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/OrderButton.js (hunk 2 above)
Comment (smottal) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/OrderButton.js]: Done

File: modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemsList.js (hunk 3 above)
Comment (smottal) [reply modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/mini_cart/CartItemsList.js]: Done

File: modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/frontend/data/set/provider/PreviewCommerceOrderItemFDSDataProvider.java (hunk 4 above)
Comment (smottal) [reply modules/apps/commerce/commerce-order-content-web/src/main/java/com/liferay/commerce/order/content/web/internal/frontend/data/set/provider/PreviewCommerceOrderItemFDSDataProvider.java:141]: Fixed

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java (hunk 5 above)
Comment (smottal) [reply modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-catalog-impl/src/main/java/com/liferay/headless/commerce/delivery/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java]: Not needed anymore for [this](https://github.com/liferay-commerce/liferay-portal/blob/bafacf008add38955f7c1f00eb3ef43ac30f2a8b/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/price/BaseCommerceProductPriceCalculation.java#L629)

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.

Thank you!

**Comment by baharturk** [comment]
ci:test:sf

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:commerce-functional

**Comment by baharturk** [comment]
ci:close

**Pull request closed without merging**